server/src/modules/response/response.service.ts

```diff
@@ -5,28 +5,61 @@ import { SlideResponseObject } from '../../shared/interfaces/response-objects/Sl
 import { SuperObjectDocument } from '../../schemas/object/super.object.schema';
 import { ObjectResponseObject } from '../../shared/interfaces/response-objects/ObjectResponseObject';
 import { BoardResponseObject } from '../../shared/interfaces/response-objects/BoardResponseObject';
+import { BoardPermissions } from 'src/shared/interfaces/BoardPermissions';
+import { BoardPermission } from 'src/enums/board.permission';
+import { Types } from 'mongoose';
+import { BoardsPaginatedResponseObject } from 'src/shared/interfaces/response-objects/BoardsPaginatedResponseObject';
 
 @Injectable()
 export class ResponseService {
   constructor() {}
 
-  toResponseBoard(board: SuperBoardDocument): BoardResponseObject {
-    const { _id, name, owner, permissions, slides } =
+  toResponseBoard(
+    board: SuperBoardDocument,
+    userId: string,
+  ): BoardResponseObject {
+    const { _id, name, owner, permissions, slides, createdAt, updatedAt } =
       board.toObject<SuperBoardDocument>();
 
     const newSlides = slides.map(slide => slide._id as string);
+    const currentUserPermission =
+      BoardPermission[this.getCurrentUserPermission(permissions, userId)];
+    const size = 1;
 
     return {
       _id: _id as string,
       name,
       owner,
       permissions,
       slides: newSlides,
+      createdAt,
+      updatedAt,
+      currentUserPermission,
+      size,
+    };
+  }
+
+  toResponseBoardsPaginated(
+    boards: SuperBoardDocument[],
+    userId: string,
+    currentPage: number,
+    perPage: number,
+    totalPages: number,
+  ): BoardsPaginatedResponseObject {
+    const boardsResponse = boards.map(board =>
+      this.toResponseBoard(board, userId),
+    );
+
+    return {
+      boardsPaginated: boardsResponse,
+      currentPage,
+      perPage,
+      totalPages,
     };
   }
 
   toResponseSlide(slide: SuperSlideDocument): SlideResponseObject {
-    const { _id, objects, version } = slide as SuperSlideDocument;
```

Comment:
It should stay `slide as SuperSlideDocument;` - otherwise it throws error when fetching slide... If you didnt modify it on purpose - i think your IDE probably changes it automatically as mine - idk how to fix that for now